sources/react-app/src/pages/Post.js

```diff
@@ -14,22 +14,73 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import BaseLayout from '../shared/BaseLayout';
 import ContentType from '../shared/ContentType';
-import PopularPostsAside from '../shared/PopularPostsAside';
+import RecentPostsAside from '../shared/RecentPostsAside';
 import PostCard, { IMAGE_BACKGROUND } from '../shared/PostCard';
-import SidebarCategories from '../shared/SidebarCategories';
-import SidebarTags from '../shared/SidebarTags';
+import { SidebarCategories, SidebarTags } from '../shared/SidebarTaxonomies';
 import SidebarSearch from '../shared/SidebarSearch';
 import SidebarBios from '../shared/SidebarBios';
 import DropZone from '../shared/DropZone';
+import { fetchQuery } from '../relayEnvironment';
+import { parseDescriptor } from '@craftercms/content';
+import ReactPaginate from 'react-paginate';
+import Comments, { CommentsCount } from '../shared/Comments';
+import { postsQuery } from '../shared/queries.graphql';
+import { useIntl } from 'react-intl';
 
 export default function (props) {
-  const { model, posts } = props;
+  const {
+    model,
+    posts,
+    meta: {
+      siteTitle,
+      socialLinks,
+      disqus: {
+        websiteShortname
+      }
+    }
+  } = props;
+  const { formatDate } = useIntl();
   const modelPath = model.craftercms.path;
+  const [related, setRelated] = useState();
+  const [totalRelatedPosts, setTotalRelatedPosts] = useState(0);
+  const [pageCount, setPageCount] = useState(0);
+  const [paginationData, setPaginationData] = useState({
+    itemsPerPage: 10,
+    currentPage: 0
+  });
+
+  useEffect(() => {
+    const categoriesFilter = model.categories_o?.map(category => {
+      return { matches: category.key }
+    });
+    const tagsFilter = model.tags_o?.map(tag => {
+      return { matches: tag.key }
+    });
+
+    fetchQuery(
+      { text: postsQuery },
+      {
+        limit: paginationData.itemsPerPage,
+        offset: (paginationData.currentPage * paginationData.itemsPerPage),
+        exclude: model.craftercms.path,
+        categoriesFilter,
+        tagsFilter
+      }
+    ).then(({ data }) => {
+      setTotalRelatedPosts(data.posts.total);
+      setRelated(parseDescriptor(data.posts.items));
```

Comment:
Why separate states for props that change together?
Though this is not a complete duplicate, it may still fall under the same refactor as the other places.